xonsh/base_shell.py

```diff
@@ -339,8 +339,12 @@ def default(self, line):
                 hist.last_cmd_rtn = 1  # return code for failure
         finally:
             ts1 = ts1 or time.time()
-            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee.getvalue())
+            tee_out = tee.getvalue()
+            self._append_history(inp=src, ts=[ts0, ts1], tee_out=tee_out)
             self.accumulated_inputs += src
+            append_newline = env.get('XONSH_APPEND_NEWLINE')
+            if append_newline and not tee_out.endswith(os.linesep):
+                print(append_newline, end='')
```

Comment:
sorry it is `events.on_postcommand`, yeah you can check various attributes of the last command.
try:
`print(events.on_postcommand.__doc__)`

i think we should just print a newline and leave more complicated cases for the events